components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/ActionManagementServiceImpl.java

```diff
@@ -268,4 +355,83 @@ private Action updateEndpointAuthenticationProperties(String actionType, String
+     * @throws ActionMgtException if action model is invalid.
+     */
+    private void doPreAddActionValidations(String actionType, Action action) throws ActionMgtException {
+
+        if (actionType.equals(Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getActionType())) {
+            ActionManagementUtil.isFieldEmpty(action.getName());
+            ActionManagementUtil.isFieldEmpty(action.getEndpoint().getUri());
+            ActionManagementUtil.isValidActionName(action.getName());
+            ActionManagementUtil.isValidEndpointUri(action.getEndpoint().getUri());
+            // Validate endpoint authentication details in the request.
+            Authentication endpointAuthentication = action.getEndpoint().getAuthentication();
+            String endpointAuthenticationType = endpointAuthentication.getType().getName();
+            doEndpointAuthenticationValidation(endpointAuthenticationType, endpointAuthentication);
+
+        }
+    }
+
+    /**
+     * Perform pre validations on action model when updating an existing action.
+     * This is specifically used during HTTP PATCH operation and
+     * only validate non-null and non-empty fields.
+     *
+     * @param actionType Action type.
+     * @param action Action update model.
+     * @throws ActionMgtClientException if action model is invalid.
+     */
+    private void doPreUpdateActionValidations(String actionType, Action action) throws ActionMgtClientException {
+
+        if (actionType.equals(Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getActionType())) {
+            // Validate the action name if present.
+            if (action.getName() != null) {
+                ActionManagementUtil.isValidActionName(action.getName());
+            }
+            // Validate the endpoint URI if present.
+            if (action.getEndpoint() != null && action.getEndpoint().getUri() != null) {
+                ActionManagementUtil.isValidEndpointUri(action.getEndpoint().getUri());
+            }
+            // Validate the endpoint authentication if present.
+            if (action.getEndpoint() != null && action.getEndpoint().getAuthentication() != null) {
+                Authentication endpointAuthentication = action.getEndpoint().getAuthentication();
+                String endpointAuthenticationType = endpointAuthentication.getType().getName();
+                doEndpointAuthenticationValidation(endpointAuthenticationType, endpointAuthentication);
+            }
+        }
+    }
+
+    /**
+     * Perform pre validations on endpoint authentication model.
+     *
+     * @param authenticationType Authentication type.
+     * @param authentication     Endpoint authentication model.
+     * @throws ActionMgtClientException if endpoint authentication model is invalid.
+     */
+    private void doEndpointAuthenticationValidation(String authenticationType, Authentication authentication)
+            throws ActionMgtClientException {
+
+        if (StringUtils.isNotBlank(authenticationType) &&
```

Comment:
Bring this common condition check to top `StringUtils.isNotBlank(authenticationType)`.
```suggestion
        if (StringUtils.isBlank(authenticationType)) {
            return;
        }
```

Reply:
Updated with 32dd909177a2b964421041250c0086df4c9737ba